src/main/java/com/fasterxml/jackson/databind/deser/BasicDeserializerFactory.java

```diff
@@ -1704,11 +1704,18 @@ public JsonDeserializer<?> createEnumDeserializer(DeserializationContext ctxt,
 
             // Need to consider @JsonValue if one found
             if (deser == null) {
-                deser = new EnumDeserializer(constructEnumResolver(enumClass,
-                        config, beanDesc.findJsonValueAccessor()),
+                Class<?> mixInClass = config.findMixInClassFor(enumClass);
```

Comment:
Ok, no. I'd prefer that mix-in handling does not get handle all over the place: they really should only be processed to produce `AnnotatedClass`, `AnnotatedMethod` etc. Code downstream should not be aware of mix-ins at all.